src/Common/MessageBroker/Integration/AmqpLib/AmqpConsumer.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\MessageBroker\Integration\AmqpLib;
+
+use ArrayObject;
+use Gaming\Common\MessageBroker\Consumer;
+use Gaming\Common\MessageBroker\Exception\MessageBrokerException;
+use Gaming\Common\MessageBroker\Integration\AmqpLib\ConnectionFactory\ConnectionFactory;
+use Gaming\Common\MessageBroker\Integration\AmqpLib\Event\MessageReturned;
+use Gaming\Common\MessageBroker\Integration\AmqpLib\MessageRouter\MessageRouter;
+use Gaming\Common\MessageBroker\Integration\AmqpLib\MessageTranslator\MessageTranslator;
+use Gaming\Common\MessageBroker\Integration\AmqpLib\QueueConsumer\QueueConsumer;
+use Gaming\Common\MessageBroker\Integration\AmqpLib\QueueConsumer\CallbackFactory;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Throwable;
+
+/**
+ * @template T
+ * @implements Consumer<QueueConsumer>
+ */
+final class AmqpConsumer implements Consumer
+{
+    private bool $shouldStop;
+
+    /**
+     * @var ArrayObject<int, AmqpContext>
+     */
+    private readonly ArrayObject $pendingMessageToContext;
+
+    public function __construct(
+        private readonly ConnectionFactory $connectionFactory,
+        private readonly int $prefetchCount,
+        private readonly MessageRouter $messageRouter,
+        private readonly MessageTranslator $messageTranslator,
```

Comment:
Just a small inconsistency, the `AmqpPublisher` first defines the `MessageTranslator` and then the `MessageRouter`. That should be aligned.